File: backend/src/main/java/com/expensesplitter/expense/dto/ExpenseParticipantResponse.java
package com.expensesplitter.expense.dto;

import lombok.*;

import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ExpenseParticipantResponse {

    private Long userId;
    private String userName;

    private BigDecimal amountOwed;
    private BigDecimal amountPaid;
    private BigDecimal netBalance;
}

File: android/app/src/main/java/com/expensesplitter/app/model/ExpenseParticipantResponse.kt
package com.expensesplitter.app.model

import java.math.BigDecimal

data class ExpenseParticipantResponse(
    val userId: Long,
    val userName: String,
    val amountOwed: BigDecimal,
    val amountPaid: BigDecimal,
    val netBalance: BigDecimal
)
